File: code-example-tests/java/driver-sync/src/main/java/getStarted/GetStarted.java
//	:replace-start: {
//	  "terms": {
//      "class GetStarted": "class QuickStart",
//      "public Document runGetStarted()": "public static void main(String[] args)",
//      "System.getenv(\"CONNECTION_STRING\")": "\"<connection string uri>\""
//	  }
//	}
package getStarted;

// :snippet-start: example
import static com.mongodb.client.model.Filters.eq;

import org.bson.Document;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;

public class GetStarted {
    public Document runGetStarted() {

        // Replace the placeholder with your MongoDB deployment's connection string
        String uri = System.getenv("CONNECTION_STRING");

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");

            Document doc = collection.find(eq("title", "Back to the Future")).first();
            if (doc != null) {
                System.out.println(doc.toJson());
                return doc; // :remove:
            } else {
                System.out.println("No matching documents found.");
                return null; // :remove:
            }
        }
    }
}
// :snippet-end:
// :replace-end:

File: code-example-tests/go/driver/examples/get_started/get_started.go
//		:replace-start: {
//		  "terms": {
//	     	"utils.GetConnectionString()": "os.Getenv(\"MONGODB_URI\")",
//			"package get_started": "package main"
//		  }
//		}
//
// :snippet-start: example
package get_started

import (
	"context"
	"encoding/json"
	"fmt"
	"log"
	//:uncomment-start:
	//"os"
	//:uncomment-end:
	"driver-examples/utils" //:remove:

	"go.mongodb.org/mongo-driver/v2/bson"
	"go.mongodb.org/mongo-driver/v2/mongo"
	"go.mongodb.org/mongo-driver/v2/mongo/options"
)

// :uncomment-start:
// func main() {
// :uncomment-end:
func GetStarted() []bson.M { //:remove:
	uri := utils.GetConnectionString()
	docs := "www.mongodb.com/docs/drivers/go/current/"
	if uri == "" {
		log.Fatal("Set your 'MONGODB_URI' environment variable. " +
			"See: " + docs +
			"usage-examples/#environment-variable")
	}
	client, err := mongo.Connect(options.Client().
		ApplyURI(uri))
	if err != nil {
		panic(err)
	}

	defer func() {
		if err := client.Disconnect(context.TODO()); err != nil {
			panic(err)
		}
	}()

	coll := client.Database("sample_mflix").Collection("movies")
	title := "Back to the Future"

	var result bson.M
	err = coll.FindOne(context.TODO(), bson.M{"title": title}).
		Decode(&result)
	if err == mongo.ErrNoDocuments {
		fmt.Printf("No document was found with the title %s\n", title)
		return []bson.M{} // :remove:
		// :uncomment-start:
		// return
		// :uncomment-end:
	}
	if err != nil {
		panic(err)
	}

	jsonData, err := json.MarshalIndent(result, "", "    ")
	if err != nil {
		panic(err)
	}
	fmt.Printf("%s\n", jsonData)
	return []bson.M{result} // :remove:
}

// :snippet-end:
// :replace-end:
